chaco/base.py

```diff
@@ -226,4 +245,57 @@ def point_line_distance(pt, p1, p2):
     return sqrt(dot(diff,diff))
 
 
-#EOF
+def intersect_range(x, low, high, mask=None):
+    """ Discard 1D intervals outside of range, with optional mask
+
+    This is an optimized routine for detecting which points are endpoints
+    of visible segments in a 1D polyline.  An optional mask can be provided for
+    points which should be excluded from consideration for other reasons
+    (such as not being selected).  Returns a mask of points which are
+    endpoints of intervals which potentially intersect the range.
+
+    Parameters
+    ----------
+    x : 1d array
```

Comment:
I'm having some trouble understanding how the entries of `x` are related to the intervals. If `x = [A, B, C, ...]`, are the intervals `[A, B], [B, C], ...`?